homeassistant/components/geniushub/__init__.py

```diff
@@ -130,11 +226,13 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 
     async_track_time_interval(hass, broker.async_update, SCAN_INTERVAL)
 
-    for platform in PLATFORMS:
-        hass.async_create_task(async_load_platform(hass, platform, DOMAIN, {}, config))
-
     setup_service_functions(hass, broker)
 
+    for platform in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(config, platform)
+        )
```

Comment:
There's a better way for this
```suggestion
    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
```